src/components/map/layers/NodeMarker.tsx

```diff
@@ -0,0 +1,152 @@
+import React, { Component } from 'react';
+import { Marker, Circle } from 'react-leaflet';
+import * as L from 'leaflet';
+import { observer, inject } from 'mobx-react';
+import classnames from 'classnames';
+import { IStop } from '~/models';
+import NodeType from '~/enums/nodeType';
+import PinIcon from '~/icons/pin';
+import { MapStore } from '~/stores/mapStore';
+import * as s from './nodeMarker.scss';
+
+// The logic of Z Indexes is not very logical.
+// Setting z-index to 2, if other items is 1 wont force it to be on top.
+// Setting z-index to a very high number will however most likely set the item on top.
+// https://leafletjs.com/reference-1.3.4.html#marker-zindexoffset
+const VERY_HIGH_Z_INDEX = 1000;
+
+interface INodeMarkerProps {
+    nodeType: NodeType;
+    latLng: L.LatLng;
+    isSelected?: boolean;
+    color?: string;
+    onContextMenu?: Function;
+    onClick?: Function;
+    isDraggable?: boolean;
+    stop?: IStop;
+    mapStore?: MapStore;
+}
+
+const DEFAULT_RADIUS = 25;
+const HASTUS_MIN_ZOOM = 16;
+
+@inject('mapStore')
+@observer
+export default class NodeMarker extends Component<INodeMarkerProps> {
+    static defaultProps = {
+        isSelected: false,
+        isDraggable: false,
+    };
+
+    private renderMarkerLabel = () => {
```

Comment:
Rather have it take stop and zoom as parameters. Would be cleaner in my opinion when called with those values from render.